lib/pbench/server/__init__.py

```diff
@@ -65,6 +65,67 @@ def tstos(ts: float = None) -> str:
     return dt.strftime("%Y-%m-%dT%H:%M:%S-%Z")
 
 
+def _get_valid_path(
+    env_name: str, dir_val: str, logger: Optional[Logger]
+) -> Optional[Path]:
+    """Get the fully resolved path from the given path.
+
+    If a logger is given, will emit error logs when the directory is not
+    found or the resolved name is not a directory.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object of the directory on disk, None otherwise.
+    """
+    try:
+        dir_path = Path(dir_val).resolve(strict=True)
+    except FileNotFoundError:
+        if logger:
+            logger.error(
+                "The {} directory, '{}', does not resolve to a real location",
+                env_name,
+                dir_val,
+            )
+        return None
+    else:
+        if not dir_path.is_dir():
+            if logger:
+                logger.error(
+                    "The {} directory, does not resolve to a directory ('{}')",
+                    env_name,
+                    dir_path,
+                )
+            return None
+    return dir_path
+
+
+def get_resolved_dir(
+    env_name: str, dir_val: str, logger: Optional[Logger]
+) -> Optional[Path]:
+    """Get the resolved directory associated with the environment name.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object, None if the directory value does not
+            resolve to a real directory.
+    """
+    dir_path = _get_valid_path(env_name, dir_val, logger)
+    if not dir_path:
+        return None
+
+    return dir_path
```

Comment:
Does this code accomplish anything?

The only times that `not dir_path` is `True` is when `dir_path` is `None`, so line 126 would produce the same result if we removed lines 123-125.  And, so `get_resolved_dir()` appears to be a straight pass-through for `_get_valid_path()` (i.e., it it doesn't rename or reorder the parameters).

Can't we just rename `_get_valid_path()` to `get_resolved_dir()` and remove this function?

Reply:
> Can't we just rename `_get_valid_path()` to `get_resolved_dir()` and remove this function?

It looks like we can.

Reply:
Yes, but _should_ we?  _**Will**_ we??  (Inquiring minds want to know! 😆)

Reply:
No ad revenue if I give away the ending.

Reply:
:rofl: